### ISSUE TITLE
Meteor output on server restart breaks in WebStorm 11.0.2 under Windows

### ISSUE THREAD

**Issue by pcarrier** [meteor/meteor#5798]
Environment:
- Meteor 1.2.1 (official Windows installer)
- WebStorm 11.0.2
- Windows 10 Pro US

Meteor seems to output some sort of control characters(?) that the WebStorm output pane doesn't care for, eg:

```
=> Server modified -- restarting...                                   => Meteor server restarted
```

In action:

![image](https://cloud.githubusercontent.com/assets/8641/11774956/69204fea-a1ef-11e5-90bc-2a381177c65a.png)

**Comment by pcarrier** [comment]
Those are 0x08 (backspaces).

**Comment by reel** [comment]
The problem is with runnerw.exe which wraps the output from cmd. You can easily rely on Terminal to run meteor (which outputs fine).

This is a webstorm problem :smile:

**Comment by pcarrier** [comment]
Well, is it a reasonable assumption that backspaces would reliably remove existing output from a log? (It intuitively isn't for me if I'm debugging an issue)

If not, should we be able to determine whether backspaces should be used or not?

On POSIX `isatty` seems like an obvious good choice, I'm not sure what exists if anything in the Windows world, or whether runnerw would honor the convention(?).

**Comment by reel** [comment]
Works on most terminals...

runnerw is mostly just a wrapper (sending terminal output and catching
control sigs) where the problem sits (i tried swapping runnerw.exe with
cmd.exe and it fails).

Backspaces are used to rewrite the spinner symbol. Try running meteor in
the WebStorm terminal, it will work fine.

You can try a simple nodejs app with blessed and the output will be garbled
also.

Have a look at
https://github.com/JetBrains/intellij-community/tree/master/native/runner/runnerw
-- this is a Webstorm issue.


Félix

**Comment by pcarrier** [comment]
I feel like we have a mismatch in expectations here.

Why would meteor expect to be outputting to a terminal, and have control over the output?

I don't see Webstorm treating the process output as append-only as a "bug".

If anything that's a feature, as garbage output shouldn't be able to hide messages.

**Comment by stubailo** [comment]
Does this actually break running the app, or does it just look ugly?

**Comment by pcarrier** [comment]
@stubailo It's just visual. Without a really, really high window size or line wrapping, one cannot see the "=> Meteor server restarted" message.

**Comment by stubailo** [comment]
I think @justinsb has a good handle on where the code is that handles this. We used to have a way to turn off the progress bar, perhaps we should add that back in or have some other solution.

**Comment by abernix** [comment]
Tagging as related to #6785 & #8082.  #8082 was opened/merged to fix this for PowerShell, but would likely still break in this case.

Two options:
* Instead of checking `!== "Windows PowerShell"` it may be possible to check `=== "Command Prompt"` (or whatever the title is in Windows).  I think so long as `\r` was used, this would work properly.
* If that doesn't work, my https://github.com/abernix/meteor/commit/ba49ff53dc51cc78389e6c055e373c701e01125c may be another option.

If someone has an easy way to test either of those solutions, please do (I do not).

**Comment by hwillson** [comment]
It looks like this issue was addressed by the merged #5816 PR. Closing - thanks!
